File: onlineShopping/Entities/OrderAggregate/Order.cs
using System.ComponentModel.DataAnnotations;

namespace onlineShopping.Entities.OrderAggregate
{
    public class Order
    {
        public int Id { get; set; }

        public string BuyerId { get; set; }
        [Required]
        public ShippingAddress ShippingAddress { get; set; }

        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        public List<OrderItem> OrderItems { get; set; }

        public decimal Subtotal { get; set; }

        public decimal DeliveryFee { get; set; }

        public PaymentStatus PaymentStatus { get; set; }
        public string OrderStatus { get; set; }
        public string PaymentIntendId { get; set; }


        public decimal GetTotal()
        {
            return Subtotal + DeliveryFee;
        }



    }
}

File: client/src/app/models/order.ts
﻿export interface Order {
    id: number;
    buyerId: string;
    shippingAddress: ShippingAddress;
    orderDate: string;
    orderItems: OrderItem[];
    subtotal: number;
    deliveryFee: number;
    orderStatus: string;
    paymentStatus:string;
    total: number;
}

export interface ShippingAddress {
    fullName: string;
    address1: string;
    address2: string;
    city: string;
    state: string;
    postal_code: string;
    country: string;
}

export interface OrderItem {
    productId: number;
    nameEn: string;
    nameAr: string;
    pictuerUrl: string;
    price: number;
    quantity: number;
}
